### ISSUE TITLE
Demo App Crashes

### ISSUE THREAD

**Issue by zeeshanz** [aws/amazon-chime-sdk-ios#311]
Followed the steps to install the demo app. Installed AmazonChimeSDKMedia.framework 16.2

As soon as it tries to run, it crashes.

<img width="1440" alt="Screen Shot 2021-06-15 at 5 48 31 PM" src="https://user-images.githubusercontent.com/4508321/122129537-418eca00-ce04-11eb-8b8e-5de9469052fb.png">

**Comment by zhinang-amazon** [comment]
Hi @zeeshanz, during your set up, did you unzip and copy the `AmazonChimeSDKMedia.framework` to `AmazonChimeSDK` folder.

**Comment by zeeshanz** [comment]
@zhinang-amazon Of course. Otherwise it won't compile. Here is how my project folder looks like.

<img width="326" alt="Screen Shot 2021-06-16 at 1 58 57 PM" src="https://user-images.githubusercontent.com/4508321/122269259-009dc080-ceab-11eb-8dd6-438aa962eaed.png">

**Comment by zhinang-amazon** [comment]
What Xcode version you are using and what's the iOS version on your device?

**Comment by zeeshanz** [comment]
Xcode 12.5 and iOS 14.3. I started getting this issue after updating Xcode from 12.2 to 12.5.

**Comment by zhinang-amazon** [comment]
Hi @zeeshanz can you try using `AmazonChimeSDKMedia.xcframework` instead?

**Comment by zeeshanz** [comment]
> Hi @zeeshanz can you try using `AmazonChimeSDKMedia.xcframework` instead?

Tried that. Same error:

<img width="1177" alt="Screen Shot 2021-06-18 at 1 46 09 PM" src="https://user-images.githubusercontent.com/4508321/122599306-20b0b980-d03c-11eb-8f97-199f2f4fedc1.png">

**Comment by zhinang-amazon** [comment]
I just tried a fresh workspace on a MacBook Pro (intel chip) with MacOS 11.3.1 BigSur and was able to run the demo app from Xcode 12.5 onto a iOS 14.4.2 iphone 7

Have you changed any build setting on the demo app project?
What mac machine are you building from?

**Comment by zeeshanz** [comment]
@zhinang-amazon It is MacBook Pro (intel chip) with MacOS 11.4 BigSur with Xcode 12.5 onto a iOS 14.3 iPhone 7 Plus. Yes, build settings were changed as described in the instructions.

I'll try again with a brand new workspace later today and see if it makes any difference.

**Comment by zeeshanz** [comment]
@zhinang-amazon So I managed to compile and run it in a brand new workspace. I had to do few things which are not mentioned in the instructions. E.g. switching Validate Workspace back and forth to Yes and No for the Project and all the Targets. Though it is not clear for which one of these it actually works and doesn't work and why it needs to be done. Then there should be better explanation of which exactly the Framwork folder is in use, since there are two references to the AmzonChimeSDK project, one of which is inside the AmazonChimeSDKDemo project. Why is that? The demo code seems to be already set with Build Settings and there should be no need to repeat these settings in the instructions.

And then how to deal with Signing & Capabilities Settings, is not explained anywhere. There is bunch of stuff here, for the project itself and each of the targets which you don't know what do with. What are these App Groups? Do we need them? If yes then how do we edit them for our use? And of course Bundle Identifiers need to be changed but which ones? And what is this AmazonChimeDemoSDKBroadcast.appex?

The instructions on how to install and use this demo app needs updating. Currently there is lots of guess work there which just wastes time. A good idea is to make a screen capture of showing how to set it up, it'll save everyone's time.

**Comment by zhinang-amazon** [comment]
Thanks @zeeshanz for the feedback. 
I don't think toggling `Validate Workspace` setting is necessary to compile and run the demo app. And our instructions around App Groups and Broadcast App extension target is linked from step 4 to the [Content Share](https://github.com/aws/amazon-chime-sdk-ios/blob/master/guides/content_share.md) guide.

However, I do think we can improve our README and set up instructions to include what builders need to update based on their own developer account. I will mark this issue as documentation and our team will work on updating README based on your feedback.

**Comment by zeeshanz** [comment]
In the end, after removing all the overblown code, it looks like this. Much simpler, and could have saved tons of time and headache.

<img width="1169" alt="Screen Shot 2021-06-21 at 9 41 52 AM" src="https://user-images.githubusercontent.com/4508321/122772547-7fad4300-d275-11eb-90f5-e20673938464.png">

**Comment by ishwarATanveshak** [comment]
Hey @zhinang-amazon and @zeeshanz, I am currently wondering about what are **broadcastBundleId** and **appGroupId**. I am not able to get it properly What are they? and where do i get them from? 
<img width="896" alt="Screenshot 2022-01-19 at 12 34 23 PM" src="https://user-images.githubusercontent.com/88021875/150081942-4fb0f6f3-6b7d-4c4e-aac5-16738b8caa3f.png">
As per my understanding app appGroupId can be set from xcode but what about **broadcastBundleId**. I am not able to configure it properly. Will you please help me over that. 
Thanks in advance.
